src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Utils/SPDXToSbomFormatConverterExtensions.cs

```diff
@@ -185,17 +185,32 @@ private static string GetSingleLicense(this Element element, RelationshipType re
 
     private static string GetSupplier(this Package spdxPackage, List<Element> spdx30Elements)
     {
-        var organizationSpdxId = spdxPackage.SuppliedBy;
+        var organizationSpdxId = spdxPackage.GetOrganizationSpdxId(spdx30Elements);
         if (organizationSpdxId is null)
         {
             return null;
         }
 
         var organizationElement = spdx30Elements
-            .FirstOrDefault(element => element is Organization && element.SpdxId == organizationSpdxId);
+            .FirstOrDefault(element => element.SpdxId == organizationSpdxId);
         return organizationElement?.Name;
     }
 
+    private static string GetOrganizationSpdxId(this Package spdxPackage, List<Element> spdx30Elements)
+    {
+        // Handle special case for the root package.
+        if (spdxPackage.SpdxId == Parsers.Spdx30SbomParser.Constants.RootPackageIdValue)
+        {
+            var creationInfo = spdx30Elements
+            .First(element => element is CreationInfo) as CreationInfo;
+            return creationInfo.CreatedBy.First();
```

Comment:
My first thought here was "Can `creationInfo` ever be null?" Searching for callers of this, I find only `SPDXToSbomFormatConverterExtensions.ToSbomPackage`, which only seems to be called from test code.

Are there customer scenarios where this would get called? It's ok if the answer is no, but I'm a bit curious about the intent of the change

Reply:
The intent of this change is so that SPDX 3.0 elements can be converted to a standardized internal format, much like the SPDX 2.2 extensions that are a mirror of this. These extensions are then used in the SBOM diff checker - https://github.com/microsoft/sbom-tool/pull/1011 so that a given SPDX 2.2 file and SPDX 3.0 file can be compared based on content. 

Currently this would never be called by the customer.

I am rethinking this change being part of this PR though. Would it be helpful if I removed the extensions change as part of this PR?

Reply:
I've separated the extensions changes in a separate PR - https://github.com/microsoft/sbom-tool/pull/1022. This can be reviewed later as well since it is not a release blocker.